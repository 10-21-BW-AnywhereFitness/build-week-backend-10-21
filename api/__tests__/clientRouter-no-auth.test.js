const request = require('supertest')
const server = require('../server')
const db = require('../data/db-config')

beforeAll(async () => {
  await db.migrate.rollback()
  await db.migrate.latest()
})
beforeEach(async () => {
  await db.seed.run()
})
afterAll(async () => {
  await db.destroy()
})

describe('[GET] /api/client/public/classes (no auth)', () => {
  let res
  beforeEach(async () => {
    res = await request(server).get('/api/client/public/classes')
  })
    it('does not require auth and returns status 200', () => {
      expect(res.status).toBe(200);
    })
    it('retrieves 4 classes', () => {
      expect(res.body).toHaveLength(4);
    })
    it('returns the data in the correct shape made for public requests', () => {
      expect(res.body).toMatchSnapshot();
    })
  })

  describe('Cannot access any of the authorized endpoints without a token', () => {
    it('cannot access [GET]/api/client/classes', async() => {
      const res = await request(server).get('/api/client/classes')
      expect(res.status).toBe(401)
      expect(res.body.message).toBe('Token required')
    })
    it('cannot access [GET]/client/classes/:class_id', async () => {
      const res = await request(server).get('/api/client/classes/1')
      expect(res.status).toBe(401)
      expect(res.body.message).toBe('Token required')
    })
    it('cannot [POST] /client/classes/:class_id', async () => {
      const res = await request(server).post('/api/client/classes/1').send()
      expect(res.status).toBe(401)
      expect(res.body.message).toBe('Token required')
    })
    it('cannot [GET] /client/:user_id/classes', async () => {
      const res = await request(server).get('/api/client/1/classes')
      expect(res.status).toBe(401)
      expect(res.body.message).toBe('Token required')
    })
    it('cannot [GET] /client/:user_id/classes/:class_id', async () => {
      const res = await request(server).get('/api/client/1/classes/1')
      expect(res.status).toBe(401)
      expect(res.body.message).toBe('Token required')
    })
    it('cannot [DELETE] /client/:user_id/classes/:class_id', async() => {
      const res = await request(server).delete('/api/client/1/classes/1')
      expect(res.status).toBe(401)
      expect(res.body.message).toBe('Token required')
    })
  })