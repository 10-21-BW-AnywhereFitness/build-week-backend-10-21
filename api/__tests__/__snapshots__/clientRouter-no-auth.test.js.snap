// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[GET] /api/client/public/classes (no auth) returns the data in the correct shape made for public requests 1`] = `
Array [
  Object {
    "class_date": "2021-10-15T07:00:00.000Z",
    "class_description": "Imagine bike travel but better",
    "class_name": "Ride through the Alps",
    "class_time": "09:00:00",
  },
  Object {
    "class_date": "2021-10-15T07:00:00.000Z",
    "class_description": "Now let's walk it out",
    "class_name": "Walk it Out",
    "class_time": "09:00:00",
  },
  Object {
    "class_date": "2021-10-21T07:00:00.000Z",
    "class_description": "Namaste",
    "class_name": "Relaxing Yoga",
    "class_time": "18:00:00",
  },
  Object {
    "class_date": "2021-10-31T07:00:00.000Z",
    "class_description": "Bollywood dancing!",
    "class_name": "Bangin' Bhangra",
    "class_time": "12:00:00",
  },
]
`;
